### PR TITLE
Refactor `puts_columns` to allow `$stderr.puts_columns`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

`puts_columns` currently only outputs to `$stdout`. This change makes it an instance method of the `IO` class, so it can be used with any `IO` object.

Also added `Kernel::puts_columns` which outputs to `$stdout` by default to be in line with `Kernel::puts`.

Also modified it so it doesn't output trailing whitespace anymore.

--
Fixes https://github.com/caskroom/homebrew-cask/issues/24642.

### CODE REVIEW

File: Library/Homebrew/utils/puts_columns.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require "utils/tty"
+
+module Kernel
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/puts_columns.rb]: Let's put this in another module if we move it around.

Comment (MikeMcQuaid) [inline Library/Homebrew/utils/puts_columns.rb]: But I think doing that in another PR would be best as it's tricky to review as-is and compare what's changed. Best to leave it as-is in `utils.rb` for now but 👍 to moving it after that.

Comment (reitermarkus) [inline Library/Homebrew/utils/puts_columns.rb]: It needs to be in `Kernel` to work with `IO` objects, i.e. `$stderr.puts_columns`.

Comment (MikeMcQuaid) [inline Library/Homebrew/utils/puts_columns.rb]: I see. Personally I'd rather we had a class method that took an argument and move away from monkeypatching system classes.

Comment (reitermarkus) [inline Library/Homebrew/utils/puts_columns.rb]: We could change it to only format the columns, and not output them, and let default `puts` to the rest.

```ruby
puts Formatter.columns(something)
```

Comment (MikeMcQuaid) [inline Library/Homebrew/utils/puts_columns.rb]: That could work.

File: Library/Homebrew/utils/formatter/columns.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+require "utils/tty"
+
+module Formatter
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/formatter/columns.rb]: Just stick this inside the existing Formatter class; I dislike objects across multiple files

File: Library/Homebrew/cask/spec/formatter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+require "spec_helper"
+require "utils/formatter"
+require "utils/tty"
+
+describe Formatter do
+  describe "::columns" do
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/spec/formatter_spec.rb]: Can you make this part of `brew tests` instead?

**Comment by MikeMcQuaid** [comment]
@reitermarkus Thoughts on this?

**Comment by reitermarkus** [comment]
@MikeMcQuaid, I moved the `columns` method into the existing file but left the test as-is, because we will need to merge the test suites anyways at some point.

**Comment by MikeMcQuaid** [comment]
@reitermarkus As the tests are already written and working I'll make an exception this time but definitely don't want to be adding any more Homebrew functionality tests primary in Cask tests in future.

**Comment by reitermarkus** [comment]
> definitely don't want to be adding any more Homebrew functionality tests primary in Cask tests

Then I guess I now have an incentive to merge the test suites. 😄

**Comment by MikeMcQuaid** [comment]
@reitermarkus Yes 😀

**Pull request merged**